OptimizelySDK/OptlyConfig/OptimizelyConfigService.cs

```diff
@@ -0,0 +1,226 @@
+            foreach (var featureFlag in projectConfig.FeatureFlags)
+            {
+                var featureExperimentMap = new Dictionary<string, OptimizelyExperiment>();
+                var featureVariableMap = new Dictionary<string, OptimizelyVariable>();
+                featureFlag.ExperimentIds.ForEach(exId =>
+                {
+                    foreach (var expMap in experimentsMap)
+                    {
+                        if (expMap.Value.Id == exId)
+                            featureExperimentMap.Add(expMap.Key, expMap.Value);
+                    }
+                });
+
+                featureFlag.Variables.ForEach(variable =>
+                {
+                    var optimizelyVariable = new OptimizelyVariable(variable.Id,
+                        variable.Key,
+                        variable.Type.ToString().ToLower(),
+                        variable.DefaultValue);
+                    featureVariableMap.Add(variable.Key, optimizelyVariable);
+                });
+
+                var optimizelyFeature = new OptimizelyFeature(featureFlag.Id, featureFlag.Key, featureExperimentMap, featureVariableMap);
+
+                FeaturesMap.Add(featureFlag.Key, optimizelyFeature);
+            }
+            return FeaturesMap;
+        }
+
+
+        #region Helper Methods for creating id value maps
+
+        /// <summary>
+        /// Gets Map of all FeatureVariable with respect to featureId
+        /// </summary>
+        /// <param name="projectConfig">The project config</param>
+        /// <returns>Dictionary | Dictionary of FeatureFlag key and value as list of all FeatureVariable inside it</returns>
+        private IDictionary<string, List<FeatureVariable>> GetFeatureIdVariablesMap(ProjectConfig projectConfig)
+        {
+            var featureIDVariablesMap = new Dictionary<string, List<FeatureVariable>>();
+            foreach (var featureFlag in projectConfig.FeatureFlags)
+            {
+                featureIDVariablesMap.Add(featureFlag.Id, featureFlag.Variables);
+            }
+            return featureIDVariablesMap;
+        }
+
+        /// <summary>
+        /// Gets Map of FeatureVariable with respect to featureVariableId
+        /// </summary>
+        /// <param name="projectConfig">The project config</param>
+        /// <returns>Dictionary | Dictionary of FeatureVariableId as key and value as object of FeatureVariable</returns>
+        private IDictionary<string, FeatureVariable> GetVariableIdMap(ProjectConfig projectConfig)
+        {
+            var featureIdMap = new Dictionary<string, FeatureVariable>();
+            foreach (FeatureFlag featureFlag in projectConfig.FeatureFlags)
+            {
+                var featureIdLocalMap = ConfigParser<FeatureVariable>.GenerateMap(entities: featureFlag.Variables, getKey: a => a.Id, clone: true);
+                featureIdMap = featureIdMap.Union(featureIdLocalMap).ToDictionary(k => k.Key, v => v.Value);
```

Comment:
why you are using `Union`? Looks like there is no item in featureIdMap. If that's the case, I think you need to copy featureIdLocalMap.